### ISSUE TITLE
Non tif files + run from macro problem

### ISSUE THREAD

**Issue by ekatrukha** [HohlbeinLab/FTM2#3]
Hello guys, great plugin, thanks for releasing!

A couple of additions:
1) If your input files are not in tiff format (like any other microscope format, stk,czi, etc), "Folder run"/"Select files" ignores them. Can you please add "file suffix" variable?

2) As I understood, it is impossible to run plugin from macro on the already opened file. It only works on files from disk (which, again, should be only tiff).
In the part of parsing the input string, can you please add an option, if the file is already open, ignore "source" or "file" variables and just work on already opened file?

3) When using "Record" option in ImageJ/FIJI (which a lot of people use, to later use in macros), automatic recorder of ImageJ logs "begin" variable (by the name of the field dialog, corresponding to the first frame), 
while in your macro specialized parsing routine it is called "start".
So basically it is impossible to use recorded strings (like with other plugins), you need to modify a call string to a specific format.

4) In general, there should be no need in parsing variables separately, ImageJ macro interface should take care of it, as soon as input dialog names are unique.

Cheers,
Eugene

**Comment by ekatrukha** [comment]
Ok, I will be more specific.

In a macro I want to open files myself, do some preprocessing and then run your plugin
on a currently open window ("Use Opened Image and Run").

If I activate "Record" option of ImageJ/FIJI, it logs a command
`run("Use Opened Image and Run", "window=15 begin=1 end=1000 show output=[]");`

If now I try to put it in my macro, it gives an error recognizing "begin" keyword.

> [ERROR] Keyword begin not found
Did you mean: end?

By looking at your code, I can change it to "start"
`run("Use Opened Image and Run", "window=15 start=1 end=1000 show output=[]");`
Now it gives an error:

> [ERROR] Keyword output not found
Did you mean: start?

Ok, let's remove the output, since current image is processed
`run("Use Opened Image and Run", "window=15 start=1 end=1000 show");`
Now I am faced with
> [ERROR] Argument string must contain source or file variables.

In spite of being called on the current image, it looks for a source or file.
I see that in the code it is just a check for non-empty string,
so I will provide something to it (anyway it knows it will run on the current image), let's try:
`run("Use Opened Image and Run", "source=dummy window=15 start=1 end=1000 show");`

So now it applies filter to loaded image! 
In short, if you change "Begin" Dialog name to Start (or parsing routine "start" to "begin")
and probably add `type != 3`
to this statement
```
if (source_dir.equals("") && file_string.equals("")) {
              logService.error("Argument string must contain source or file variables.");
                return DONE;
}
```
It will make incorporation of your plugin to macros much more easier :)

Cheers,
Eugene

**Comment by ekatrukha** [comment]
While I am at it,
maybe also it is a good idea to remove 
the message window "Finished Applying Faster Temporal Median."
at the end of "Use Opened Image and Run", type ==3 mode.
Since when running in macro batch mode macro will pause for a user to click "OK".
Maybe it is better to put it to "Log" window (I see that progress bar/FIJI bar is updated anyway).

Once again, great plugin, super useful, thank you .
Just some suggestions on how to make it easier to incorporate to existing macros/workflows).

Cheers,
Eugene

**Comment by DrNjitram** [comment]
Hey Eugene, thanks for the suggestions!

I am not super familiar with the record function, so I was not aware of these issues. They should not be super difficult to implement thankfully.
I was under the understanding that when running from macro ImageJ suppressed all windows, and I might have subconsciously ignored the window. 
When running from macro I can simply suppress the message. It is mostly there to inform the user that the process is done, but in macro mode, this is not relevant of course.

I will implement the fixes asap!

Cheers, 
Martijn

**Comment by ekatrukha** [comment]
Hello Martijn,

Thanks again for super quick feedback, especially on Sunday
(I hope I didn't interrupt anything).
Yes, if in macro mode, it would be nice to remove it, otherwise
in batch mode have to click OK.
Thanks in advance for your help!

All the best,
Eugene

**Comment by DrNjitram** [comment]
Should be all fixed with the latest version!
Give it a whirl!
